src/lib/core/components/visualizations/implementations/HistogramVisualization.tsx

```diff
@@ -325,6 +330,20 @@ function HistogramViz(props: VisualizationProps) {
         outputEntity={outputEntity}
         independentAxisVariable={vizConfig.xAxisVariable}
         independentAxisLabel={axisLabelWithUnit(xAxisVariable) ?? 'Main'}
+        // variable's metadata-based independent axis range with margin
+        independentAxisRange={
+          defaultIndependentRange && defaultIndependentRange != null
+            ? xAxisVariable?.type === 'date'
+              ? {
+                  min: defaultIndependentRange.min as string,
+                  max: defaultIndependentRange.max as string,
+                }
+              : {
+                  min: defaultIndependentRange.min as number,
+                  max: defaultIndependentRange.max as number,
+                }
+            : undefined
```

Comment:
This should be computed in a `useMemo` hook to prevent excessive re-renders of the `Histogram` component.

Reply:
@dmfalke useMemo() is now applied to both of them (this and below ones)